### PR TITLE
update(intersect): update to raise error only when interface is called improperly

### LINKED ISSUE #1481: warning generated despite correct syntax
gwf.modelgrid.intersect() generates the warning below, despite the correct syntax. Is this intentional? (I'd hate to "fix" it if it ain't broken) It wouldn't be a big deal except I am calling it from a lambda function in a large dataframe and it blows all other information off the screen!!

```
obs['ij'] = obs[[x_field, y_field]]. apply(lambda x: gwf.modelgrid.intersect(x[0], x[1], z=None, local=False, forgive=True), axis=1)
```

`structuredgrid.py:777: UserWarning: The interface 'intersect(self, x, y, local=False, forgive=False)' has been deprecated. Use the intersect(self, x, y, z=None, local=False, forgive=False) interface instead.`

### PR DESCRIPTION
**Pull request by jlarsen-usgs**
Closes #1481

### CODE REVIEW

File: flopy/discretization/unstructuredgrid.py [hunk 1]
```diff
@@ -538,8 +538,9 @@ def intersect(self, x, y, z=None, local=False, forgive=False):
             The CELL2D number
 
         """
-        frame_info = inspect.getframeinfo(inspect.currentframe())
-        self._warn_intersect(frame_info.filename, frame_info.lineno)
+        if instance(z, bool):
```
Comment (jdhughes-usgs) [inline flopy/discretization/unstructuredgrid.py:541]: should be `isinstance(z, bool)`

**Review by jdhughes-usgs — CHANGES_REQUESTED** [review]

**Comment by w-bonelli** [comment]
The Windows/Python3.10 CI failure looks like something I introduced in a recent PR. The cache miniconda step is trying to hash the entire flopy module to check for changes, rather than just `etc/environment.yml`. I think I did that while debugging and forgot to remove it. Sorry about that.

I can fix in a PR later today, unless you want to here @jlarsen-usgs

**Comment by jlarsen-usgs** [comment]
Updated the CI. Thanks for the help @w-bonelli

**Comment by w-bonelli** [comment]
I reran CI jobs due to [more flakiness](https://github.com/modflowpy/flopy/runs/7748574800?check_suite_focus=true#step:9:1733) in the tests for `get-modflow`. This time it looks like a different internal GitHub issue. Discussions:

- https://github.com/orgs/community/discussions/8535 (note the recent reports if you scroll to bottom)
- https://github.com/lovell/sharp-libvips/issues/121

I removed the `@flaky` marker last week after @mwtoews added internal retries, but I guess it might be worth re-marking semi-permanently in case this resurfaces. Can do so in a new PR if this goes in.

**Comment by mwtoews** [comment]
Thanks for the info @w-bonelli, it seems to be a similar transient network issue. I'll apply some more internal retry logic for this HTTP error. Not sure when I'll get this done, could be a couple of days, so feel free to re-add `@flaky` for now.

**Pull request merged**